assets/components/PersonalizeHomeModal.tsx

```diff
@@ -0,0 +1,356 @@
+/* eslint-disable react/no-unescaped-entities */
+import React from 'react';
+import {connect} from 'react-redux';
+import {partition} from 'lodash';
+import {gettext} from 'utils';
+import Modal from 'components/Modal';
+import {Input} from 'reactstrap';
+import CheckboxInput from './CheckboxInput';
+import {IUser, IUserDashboard} from 'interfaces/user';
+import {RadioButtonGroup} from 'features/sections/SectionSwitch';
+import {modalFormInvalid, modalFormValid} from 'actions';
+import {updateUser} from 'users/actions';
+import {getCurrentUser} from 'company-admin/selectors';
+import {IAgendaState} from 'agenda/reducers';
+import {ITopic} from 'interfaces/topic';
+
+interface IStateProps {
+    topics: Array<ITopic>;
+    currentUser: IUser;
+}
+
+interface IDispatchProps {
+    saveUser: (updates: Partial<IUser>) => void;
+    modalFormValid: () => void;
+    modalFormInvalid: () => void;
+}
+
+interface IOwnProps {
+    closeModal?: () => void;
+}
+
+type IProps = IStateProps & IDispatchProps & IOwnProps;
+
+interface IState {
+    selectedTopicIds: Array<string>;
+    activeSection: string;
+    searchTerm: string;
+    isLoading: boolean;
+    wireTopics: Array<ITopic>;
+    wireTopicsById: Map<string, ITopic>;
```

Comment:
We should add `Immutable.js` to the project. Then we could use an ordered map instead of having to have 2 variables for the same data that need to be kept in sycn.